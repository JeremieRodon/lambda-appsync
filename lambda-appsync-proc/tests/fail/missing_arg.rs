use lambda_appsync::{appsync_lambda_main, appsync_operation, AppsyncError, ID};

appsync_lambda_main!("../../../../schema.graphql");

// Missing argument
#[appsync_operation(mutation(createPlayer))]
async fn create_player() -> Result<Player, AppsyncError> {
    Ok(Player {
        id: ID::new(),
        name: "Test".to_string(),
        team: Team::Rust,
    })
}
