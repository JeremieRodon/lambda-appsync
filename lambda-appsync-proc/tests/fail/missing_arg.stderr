error[E0277]: the trait bound `&AppsyncEvent<Operation>: DeserializeOwned` is not satisfied
 --> tests/fail/missing_arg.rs:3:22
  |
3 | appsync_lambda_main!("../../../../schema.graphql");
  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `for<'de> Deserialize<'de>` is not implemented for `&AppsyncEvent<Operation>`, which is required by `&AppsyncEvent<Operation>: DeserializeOwned`
  |
  = help: the trait `Deserialize<'_>` is implemented for `AppsyncEvent<_>`
  = help: for that trait implementation, expected `AppsyncEvent<_>`, found `&AppsyncEvent<Operation>`
  = note: required for `&AppsyncEvent<Operation>` to implement `DeserializeOwned`
note: required by a bound in `arg_from_json`
 --> $WORKSPACE/lambda-appsync/src/lib.rs
  |
  | pub fn arg_from_json<T: DeserializeOwned>(
  |                         ^^^^^^^^^^^^^^^^ required by this bound in `arg_from_json`

error[E0061]: this function takes 1 argument but 2 arguments were supplied
 --> tests/fail/missing_arg.rs:3:1
  |
3 | appsync_lambda_main!("../../../../schema.graphql");
  | ^^^^^^^^^^^^^^^^^^^^^----------------------------^
  |                      |
  |                      unexpected argument #2 of type `&AppsyncEvent<Operation>`
  |
note: associated function defined here
 --> tests/fail/missing_arg.rs:6:30
  |
6 | #[appsync_operation(mutation(createPlayer))]
  | -----------------------------^^^^^^^^^^^^---
  = note: this error originates in the macro `appsync_lambda_main` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0061]: this function takes 2 arguments but 1 argument was supplied
 --> tests/fail/missing_arg.rs:6:1
  |
6 | #[appsync_operation(mutation(createPlayer))]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ argument #1 of type `std::string::String` is missing
  |
note: associated function defined here
 --> tests/fail/missing_arg.rs:3:1
  |
3 | appsync_lambda_main!("../../../../schema.graphql");
  | ^^^^^^^^^^^^^^^^^^^^^----------------------------^
  = note: this error originates in the attribute macro `appsync_operation` which comes from the expansion of the macro `appsync_lambda_main` (in Nightly builds, run with -Z macro-backtrace for more info)
help: provide the argument
  |
6 | #[appsync_operation(mutation(createPlayer))](/* std::string::String */, #[appsync_operation(mutation(createPlayer))])
  |                                             +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
